tests/ray_cluster_test.py

```diff
@@ -0,0 +1,88 @@
+from __future__ import print_function
+
+from absl import app
+from absl import flags
+
+from ray.tests.cluster_utils import Cluster
+
+import ray.worker
+
+import erdos.graph
+from erdos.op import Op
+from erdos.utils import frequency
+
+FLAGS = flags.FLAGS
+
+# Maps plasma store socket name -> node id
+node_directory = {}
+
+
+class MachineAOp(Op):
+    def __init__(self, name):
+        super(MachineAOp, self).__init__(name)
+
+    @staticmethod
+    def setup_streams(input_streams):
+        return []
+
+    @frequency(1)
+    def print_node(self):
+        plasma_store_socket_name = (
+            ray.worker.global_worker.plasma_client.store_socket_name)
+        node_name = node_directory.get(plasma_store_socket_name, "unknown")
+        print("MachineAOp is located on node {}".format(node_name))
+
+    def execute(self):
+        self.print_node()
+        self.spin()
+
+
+class MachineBOp(Op):
+    def __init__(self, name):
+        super(MachineBOp, self).__init__(name)
+
+    @staticmethod
+    def setup_streams(input_streams):
+        return []
+
+    @frequency(1)
+    def print_node(self):
+        plasma_store_socket_name = (
+            ray.worker.global_worker.plasma_client.store_socket_name)
+        node_name = node_directory.get(plasma_store_socket_name, "unknown")
+        print("MachineBOp is located on node {}".format(node_name))
+
+    def execute(self):
+        self.print_node()
+        self.spin()
+
+
+def main(argv):
+    # Set up graph
+    graph = erdos.graph.get_current_graph()
+
+    # Add operators
+    machine_a_op = graph.add(MachineAOp, name='machine_a_op',
+                            resources={"machine_a": 1})
+    machine_b_op = graph.add(MachineBOp, name='machine_b_op',
+                            resources={"machine_b": 1})
+
+    # Execute graph
+    graph.execute("ray")
+
+
+if __name__ == "__main__":
+    cluster = Cluster(initialize_head=True)
```

Comment:
We would ideally want to have a machine and cluster abstraction that is independent of the underlying execution platform. The system needs to be able to distribute the graph both using `ROS` and `Ray` cleanly.

I would envision writing a `Machine` and a `Cluster` class that gets passed to a `GraphNode` or a `Graph` respectively, and the initialization of the execution engine happens during the setup phase. (creation of custom resources and allocation of those resources per node for Ray, and setup of a ROS master on the master node and an export of the `ROS_MASTER_URI` on every other node.)